### PR TITLE
Add dockerfile

### PR DESCRIPTION
**Pull request by mattmarnien**

### CODE REVIEW

**Review by ghost — APPROVED** [review]
Glad we added the .dockerignore

**Review by townerr — APPROVED** [review]

**Review by escalonn — CHANGES_REQUESTED** [review]
dockerfile is very basic, doesn't even build the code; not ready for main branch. should use multi-stage build

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build-env
+WORKDIR /app
+
+# Copy csproj and restore as distinct layers
+COPY *.csproj ./
+RUN dotnet restore
+
+# Copy everything else and build
+COPY . ./
+RUN dotnet public -c Release -o publish
```
Comment (escalonn) [inline Dockerfile]: dotnet public

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Build runtime image
+FROM mcr.microsoft.com/dotnet/sdk:5.0 AS runtime
```
Comment (escalonn) [inline Dockerfile]: don't use the sdk image for the final base image

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,23 @@
+#base image + dependencies
+FROM mcr.microsoft.com/dotnet/sdk:5.0 AS build-env
+WORKDIR /app
+
+# Copy csproj and restore as distinct layers
+COPY *.csproj ./
+RUN dotnet restore
```
Comment (escalonn) [inline Dockerfile]: need to copy sln and csprojs, to correct places

**Review by escalonn — CHANGES_REQUESTED** [review]
if you'd tested this, it would have failed twice over. you should catch this before it comes to me

edit: i thought you'd already re-requested review from me and had two people already approve the latest here; i read the UI wrong

**Review by townerr — APPROVED** [review]

**Comment by ghost** [comment]
I was able to test it on my local environment and got a response from /weatherforecast

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+COPY  FakebookNotifications/*.sln .
+COPY FakebookNotifications/FakebookNotifications.WebApi/*.csproj ./FakebookNotifications.WebApi/
+COPY FakebookNotifications/FakebookNotifications.Testing/*.csproj ./FakebookNotifications.Testing/
+RUN dotnet restore
+
+# Copy everything else and build
+COPY . ./
+COPY FakebookNotifications/FakebookNotifications.WebApi/. ./FakebookNotifications.WebApi/
+COPY FakebookNotifications/FakebookNotifications.Testing/. ./FakebookNotifications.Testing/
+RUN dotnet publish -c Release -o publish
```
Comment (escalonn) [inline Dockerfile]: publish the startup project specifically

**Review by escalonn — CHANGES_REQUESTED** [review]
seems like the only files the dockerfile needs are down in fakebooknotifications directory. expect the build context to be that directory. move the dockerignore down there and also the dockerfile if you prefer.
if the build context is up at repo root, `.git` also needs to be dockerignored, but it's not an issue if the context moves down.

**Review by escalonn — APPROVED** [review]

**Pull request merged**